pkg/container/module.go

```diff
@@ -308,3 +308,102 @@ func UninstallContainerd(m *UninstallContainerModule) []task.Interface {
 		disableContainerd,
 	}
 }
+
+type CriMigrateModule struct {
+	common.KubeModule
+
+	Skip bool
+}
+
+func (i *CriMigrateModule) IsSkip() bool {
+	return i.Skip
+}
+
+func (p *CriMigrateModule) Init() {
+	p.Name = "CriMigrateModule"
+	p.Desc = "Cri Migrate manager"
+
+	if p.KubeConf.Arg.Type == common.Docker && p.KubeConf.Arg.Role == common.Worker {
+		p.Tasks = MigrateToDocker(p)
+	} else if p.KubeConf.Arg.Type == common.Docker && p.KubeConf.Arg.Role == common.Master {
+		p.Tasks = MigrateMToDocker(p)
+	} else if p.KubeConf.Arg.Type == common.Conatinerd && p.KubeConf.Arg.Role == common.Worker {
+		p.Tasks = MigrateToContainerd(p)
+	} else if p.KubeConf.Arg.Type == common.Conatinerd && p.KubeConf.Arg.Role == common.Master {
+		p.Tasks = MigrateMToContainerd(p)
+	} else {
+		logger.Log.Fatalf("Unsupported container runtime: %s", strings.TrimSpace(p.KubeConf.Arg.Type))
+	}
```

Comment:
What happened if the flag `role` support this `--role all`? Add a new module?
Maybe we just need to add a new `prepare` to determine if the node should be skipped.
```go
var action task.Action
switch type {
case docker:
  action =  new(MigrateSelfNodeToDocker)
case contained:
  action = new(MigrateSelfNodeToContainerd)
}

migrate = &task.RemoteTask{
		Name:     "MigrateToDocker",
		Desc:     "Migrate To Docker",
		Hosts:    p.Runtime.GetHostsByRole(common.k8s),
		Prepare:  new(RolePrepare),
		Action:   action,
		Parallel: false,
	}
```

Reply:
ok,I will optimize it,soon